Should codepaths that use java.nio catch ProviderNotFoundException?

I was running on a non-HDFS file system that did not have a registered `java.nio.file.spi.FileSystemProvider` for the FS scheme. I was loading the whole file, so I didn't _need_ the index, but was getting a `ProviderNotFoundException` when [loading the index when picking splits](https://github.com/HadoopGenomics/Hadoop-BAM/blob/master/src/main/java/org/seqdoop/hadoop_bam/BAMInputFormat.java#L172). Turns out, the BAM split picking code [only catches an IOError](https://github.com/HadoopGenomics/Hadoop-BAM/blob/master/src/main/java/org/seqdoop/hadoop_bam/BAMInputFormat.java#L152-L156) when using the BAM index. I think the correct behavior would be to also catch the `java.nio.file.ProviderNotFoundException` exception as well. Thoughts?

I think I can confirm what @fnothaft said in the last post about BAMRecordReader. This error is thrown after trying to load the entire bam file on s3. Let me know if you want to full stack trace.
```
Caused by: java.nio.file.ProviderNotFoundException: Provider "s3a" not found
  at java.nio.file.FileSystems.newFileSystem(FileSystems.java:341)
  at org.seqdoop.hadoop_bam.util.NIOFileUtil.asPath(NIOFileUtil.java:40)
  at org.seqdoop.hadoop_bam.BAMRecordReader.initialize(BAMRecordReader.java:140)
  at org.seqdoop.hadoop_bam.BAMInputFormat.createRecordReader(BAMInputFormat.java:121)
  at org.seqdoop.hadoop_bam.AnySAMInputFormat.createRecordReader(AnySAMInputFormat.java:190)
  at org.apache.spark.rdd.NewHadoopRDD$$anon$1.liftedTree1$1(NewHadoopRDD.scala:178)
  at org.apache.spark.rdd.NewHadoopRDD$$anon$1.<init>(NewHadoopRDD.scala:177)
  at org.apache.spark.rdd.NewHadoopRDD.compute(NewHadoopRDD.scala:134)
  at org.apache.spark.rdd.NewHadoopRDD.compute(NewHadoopRDD.scala:69)
  at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:323)
  at org.apache.spark.rdd.RDD.iterator(RDD.scala:287)
  at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:38)
  at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:323)
  at org.apache.spark.rdd.RDD.iterator(RDD.scala:287)
  at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:38)
  at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:323)
  at org.apache.spark.rdd.RDD.iterator(RDD.scala:287)
  at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:87)
  at org.apache.spark.scheduler.Task.run(Task.scala:99)
  at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:322)
```

--- a/src/main/java/org/seqdoop/hadoop_bam/BAMInputFormat.java
+++ b/src/main/java/org/seqdoop/hadoop_bam/BAMInputFormat.java
@@ -50,6 +50,7 @@ import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 import java.io.IOException;
+import java.nio.file.ProviderNotFoundException;
 import java.util.ArrayList;
 import java.util.Collections;
 import java.util.Comparator;
@@ -191,7 +192,7 @@ public class BAMInputFormat
 	/** Returns a {@link BAMRecordReader} initialized with the parameters. */
 	@Override public RecordReader<LongWritable,SAMRecordWritable>
 		createRecordReader(InputSplit split, TaskAttemptContext ctx)
-			throws InterruptedException, IOException
+            throws InterruptedException, IOException
 	{
 		final RecordReader<LongWritable,SAMRecordWritable> rr =
 			new BAMRecordReader();
@@ -231,7 +232,7 @@ public class BAMInputFormat
 		for (int i = 0; i < origSplits.size();) {
 			try {
 				i = addIndexedSplits      (origSplits, i, newSplits, cfg);
-			} catch (IOException e) {
+			} catch (IOException | ProviderNotFoundException e) {
 				i = addProbabilisticSplits(origSplits, i, newSplits, cfg);
 			}
 		}

--- a/src/main/java/org/seqdoop/hadoop_bam/BAMRecordReader.java
+++ b/src/main/java/org/seqdoop/hadoop_bam/BAMRecordReader.java
@@ -120,7 +120,7 @@ public class BAMRecordReader
 	}
 
 	@Override public void initialize(InputSplit spl, TaskAttemptContext ctx)
-		throws IOException
+            throws IOException
 	{
 		// This method should only be called once (see Hadoop API). However,
 		// there seems to be disagreement between implementations that call

